src/actions/UploadFileAction.ts

```diff
@@ -1,19 +1,39 @@
 'use server';
 
-import { withDatabase } from '@/database';
-import { FileController } from '@/database/controllers';
+import { revalidatePath } from 'next/cache';
+import { redirect } from 'next/navigation';
+import { withDatabase } from '../database';
+import { FileController, VideoController } from '../database/controllers';
 
-export const UploadFileAction = async (data: FormData) => {
-	'use server';
-	const file = data.get('file') as File;
+export const uploadFileAction = async (data: FormData) => {
+	const video = data.get('video') as File;
+	const image = data.get('image') as File;
+	const name = data.get('name')?.toString();
+	const description = data.get('description')?.toString();
+	const categoryId = data.get('categoryId')?.toString();
 
-	if (!file) {
-		throw new Error('No file uploaded');
+	if (!video || !image || !name || !description || !categoryId) {
+		return;
+	}
+	if (!video.type.match(/video\/(mp4|webm|ogg)/) || !image.type.match(/image\/(jpeg|png|gif|webp|svg\+xml)/)) {
+		return;
 	}
-
-	if (!file.type.match(/image\/(jpeg|png|gif|webp|svg\+xml)|video\/(mp4|webm|ogg)/)) return;
 
 	const db = await withDatabase();
 
-	await new FileController(db).upload(file);
+	const fileController = new FileController(db);
+	const uploadedVideo = await fileController.upload(video);
+	const uploadedImage = await fileController.upload(image);
+
+	const videoFile = new VideoController(db);
+
+	videoFile.create({
+		files: [uploadedImage, uploadedVideo],
+		category: { id: categoryId },
+		description: description,
+		name: name,
+	});
+
+	revalidatePath('./');
+	redirect('/');
```

Comment:
se mi moc nelíbí... prosím o vysvětlení a důvod.

Reply:
no po tom co je uploadnuty file, tak uživatele kopne na home page kde jsou fily zobrazeny

Reply:
nelíbí se mi to, je to špatně, ale zkousnu to